src/analyzer/protocol/smb/smb2-com-set-info.pac

```diff
@@ -81,13 +253,120 @@ type SMB2_file_disposition_info(sir: SMB2_set_info_request) = record {
 	proc: bool = $context.connection.proc_smb2_set_info_request_file_delete(this);
 };
 
+type SMB2_file_allocation_info(sir: SMB2_set_info_request) = record {
+	allocation_size : int64;
+} &let {
+	proc: bool = $context.connection.proc_smb2_set_info_request_file_allocation(this);
+};
+
+type SMB2_file_endoffile_info(sir: SMB2_set_info_request) = record {
+	endoffile : int64;
+} &let {
+	proc: bool = $context.connection.proc_smb2_set_info_request_file_endoffile(this);
+};
+
+type SMB2_file_fullea_info_element = record {
+	next_entry_offset : uint32;
+	flags		  : uint8;
+	ea_name_length    : uint8;
+	ea_value_length	  : uint16;
+	ea_name		  : SMB2_string(ea_name_length); 
+	ea_value	  : SMB2_string(ea_value_length);
+	pad_to_next	  : padding to next_entry_offset;
+} &let {
+	next_offset: int = next_entry_offset;
+};
+
+type SMB2_file_fullea_info(sir: SMB2_set_info_request) = record {
+	ea_vector 	  : SMB2_file_fullea_info_element[] &until($element.next_offset == 0);  
+} &let {
+	proc: bool = $context.connection.proc_smb2_set_info_request_file_fullea(this);
+};
+
+type SMB2_file_link_info(sir: SMB2_set_info_request) = record {
+	replace_if_exists : uint8;
+	reserved	  : uint8[7];
+	root_directory 	  : uint64;
+	file_name_length  : uint32;
+	file_name	  : SMB2_string(file_name_length);
+} &let {
+	proc: bool = $context.connection.proc_smb2_set_info_request_file_link(this);
+};
+
+type SMB2_file_mode_info(sir: SMB2_set_info_request) = record {
+	mode : uint32;
+} &let {
+	proc: bool = $context.connection.proc_smb2_set_info_request_file_mode(this);
+};
+
+type SMB2_file_pipe_info(sir: SMB2_set_info_request) = record {
+	read_mode 	: uint32;
+	completion_mode : uint32;
+} &let {
+	proc: bool = $context.connection.proc_smb2_set_info_request_file_pipe(this);
+};
+
+type SMB2_file_position_info(sir: SMB2_set_info_request) = record {
+	current_byte_offset 	: int64;
```

Comment:
I see you're using ints (as opposed to uints) in a lot of places here.  Should those really be ints?  It seems that a negative value for current_byte_offset would be odd.

Reply:
Well, I shouldn't have said in "a lot" of places.  I guess you're only using them in a few places.

Reply:
Some fields are signed integer as in the specs given in
[MS-FSCC] - v20180912
File System Control Codes
Copyright © 2018 Microsoft Corporation
Release: September 12, 2018